tests/unittests/unittests.cpp

```diff
@@ -4264,6 +4269,137 @@ class ConcurrentQueueTests : public TestClass<ConcurrentQueueTests>
 		
 		return true;
 	}
+
+	bool acquire_and_signal()
```

Comment:
It would be amazing if we could get the semaphore tests to run under Relacy. Maybe we could simulate the underlying OS semaphore with a C++11 concurrency primitive that Relacy understands?